### PR TITLE
Move GL utilities to their own library

### PR DESCRIPTION
**Pull request by jherico**
Currently GPU is the source of all GL dependencies, but this doesn't make sense.  GPU is our internal abstraction for rendering, but there are a number of other use cases (presentation of scene textures through the display plugins, using QML UI) where we need to work with OpenGL without the GPU abstraction.  

Many of these utility classes are simple enough that they would otherwise belong in Shared, but Shared must not link against the OpenGL libraries because of requirements for running server code against hardware that has no GL libraries.

This PR moves all these classes out of their old locations (mostly render-utils) and into a new library 'gl'.

### CODE REVIEW

File: libraries/gl/src/gl/OffscreenQmlSurface.cpp [hunk 1]
```diff
@@ -207,9 +206,7 @@ class OffscreenQmlRenderer : public OffscreenGlCanvas {
         qreal pixelRatio = 1.0;
         if (_renderControl && _renderControl->_renderWindow) {
             pixelRatio = _renderControl->_renderWindow->devicePixelRatio();
-        } else {
```
Comment (jherico) [inline libraries/gl/src/gl/OffscreenQmlSurface.cpp]: this should be fixed in 5.5.1

**Comment by hifi-gustavo** [comment]
Build console output for commit: 0c6443e509dc7d22b956058220608640f97ad364 available at: https://gist.github.com/11507260274d5d27d9a2<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6196.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6196.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 35aef5dae5b5125609ce28fdae2c900bb93028c1 available at: https://gist.github.com/11507260274d5d27d9a2<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6196.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6196.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: e78a367c110eea58f928ecdb8572a05b3b1eff9a available at: https://gist.github.com/11507260274d5d27d9a2<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6196.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6196.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: e78a367c110eea58f928ecdb8572a05b3b1eff9a available at: https://gist.github.com/11507260274d5d27d9a2<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6196.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6196.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 5cbb2562cd1b72ea77d0f60703d34a0f0365a241 available at: https://gist.github.com/11507260274d5d27d9a2<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6196.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6196.dmg

**Comment by samcake** [comment]
I did review and think it s good to go when we can merge.
I haven't test though

**Comment by ZappoMan** [comment]
testing on windows now -- if it runs, I'm merging

**Pull request merged**